oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -1755,53 +1755,178 @@ int mca_spml_ucx_team_sync(shmem_team_t team)
     return OSHMEM_ERR_NOT_IMPLEMENTED;
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_my_pe(shmem_team_t team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_team = (mca_spml_ucx_team_t *)team;
+
+    if (team == SHMEM_TEAM_WORLD) {
+        return shmem_my_pe();
+    }
+
+    return ucx_team->my_pe;
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_n_pes(shmem_team_t team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_team = (mca_spml_ucx_team_t *)team;
+
+    if (team == SHMEM_TEAM_WORLD) {
+        return shmem_n_pes();
+    }
+
+    return ucx_team->n_pes;
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_get_config(shmem_team_t team, long config_mask,
         shmem_team_config_t *config)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_team = (mca_spml_ucx_team_t *)team;
+    SPML_UCX_VALIDATE_TEAM(team);
+
+    memcpy(config, &ucx_team->config, sizeof(shmem_team_config_t));
+
+    return SHMEM_SUCCESS;
+}
+
+static inline int mca_spml_ucx_is_pe_in_strided_team(int src_pe, int start,
+        int stride, int size)
+{
+    return (src_pe >= start) && (src_pe < start + size * stride)
+           && ((src_pe - start) % stride == 0);
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_translate_pe(shmem_team_t src_team, int src_pe,
         shmem_team_t dest_team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_src_team  = (mca_spml_ucx_team_t*) src_team;
+    mca_spml_ucx_team_t *ucx_dest_team = (mca_spml_ucx_team_t*) dest_team;
+    int global_pe;
+
+    if ((src_pe == SPML_UCX_PE_NOT_IN_TEAM) || (src_team == dest_team)) {
+        return src_pe;
+    }
+
+    global_pe = ucx_src_team->start + src_pe * ucx_src_team->stride;
+
+    if (dest_team == SHMEM_TEAM_WORLD) {
+        return global_pe;
+    }
+
+    if (!mca_spml_ucx_is_pe_in_strided_team(global_pe, ucx_dest_team->start, ucx_dest_team->stride,
+                                            ucx_dest_team->n_pes)) {
+        return SPML_UCX_PE_NOT_IN_TEAM;
+    }
+
+    return (global_pe - ucx_dest_team->start) / ucx_dest_team->stride;
 }
 
-/* This routine is not implemented */
 int mca_spml_ucx_team_split_strided(shmem_team_t parent_team, int start, int
         stride, int size, const shmem_team_config_t *config, long config_mask,
         shmem_team_t *new_team)
 {
-    return OSHMEM_ERR_NOT_IMPLEMENTED;
+    mca_spml_ucx_team_t *ucx_parent_team;
+    mca_spml_ucx_team_t *ucx_new_team;
+    int parent_pe;
+    int parent_start;
+    int parent_stride;
+    int my_pe;
+
+    SPML_UCX_ASSERT(((start + size * stride) <= oshmem_num_procs()) && (start < size) && (stride > 0) && (size > 0));
```

Comment:
```size``` can be any positive value , and start is PE index in the parent team, so ```start < size``` is not required to be true.